javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -337,6 +338,12 @@
     BitSet<T> addAll(Iterable<? extends T> elements);
 
     @Override
+    default <R> SortedSet<R> collect(PartialFunction<? super T, ? extends R> partialFunction) {
+        Objects.requireNonNull(partialFunction, "partialFunction is null");
+        return filter(partialFunction::isDefinedAt).map(partialFunction::apply);
```

Comment:
should be modified to something like `TreeSet.ofAll(iterator().filter().map())` to avoid creating temporary `BitSet`